rtengine/spot.cc

```diff
@@ -0,0 +1,513 @@
+                }
+                ++srcImgX;
+                ++dstImgX;
+            }
+            ++srcImgY;
+            ++dstImgY;
+        }
+
+        return true;
+    }
+
+    // Copy the intersecting part
+    bool copyImgTo(SpotBox &destBox) {
+        Imagefloat *destImg = destBox.image;
+
+        if (image == nullptr || destImg == nullptr) {
+            std::cerr << "One of the source or destination SpotBox image is missing !" << std::endl;
+            return false;
+        }
+
+        std::unique_ptr<Rectangle> intersection;
+
+        if (!intersectionArea.getIntersection(destBox.intersectionArea, intersection)) {
+            return false;
+        }
+
+        Imagefloat *srcImg = image;
+        Imagefloat *dstImg = destBox.image;
+
+        int srcImgY = intersection->y1 - imgArea.y1;
+        int dstImgY = intersection->y1 - destBox.imgArea.y1;
+        for (int y = intersection->y1; y <= intersection->y2; ++y) {
+            int srcImgX = intersection->x1 - imgArea.x1;
+            int dstImgX = intersection->x1 - destBox.imgArea.x1;
+
+            for (int x = intersection->x1; x <= intersection->x2; ++x) {
+                dstImg->r(dstImgY, dstImgX) = srcImg->r(srcImgY, srcImgX);
+                dstImg->g(dstImgY, dstImgX) = srcImg->g(srcImgY, srcImgX);
+                dstImg->b(dstImgY, dstImgX) = srcImg->b(srcImgY, srcImgX);
+                ++srcImgX;
+                ++dstImgX;
+            }
+            ++srcImgY;
+            ++dstImgY;
+        }
+
+        return true;
+    }
+};
+
+void ImProcFunctions::removeSpots (Imagefloat* img, ImageSource* imgsrc, const std::vector<SpotEntry> &entries, const PreviewProps &pp, const ColorTemp &currWB, const ColorManagementParams *cmp, int tr)
+{
+    //Get the clipped image areas (src & dst) from the source image
+
+    std::vector< std::shared_ptr<SpotBox> > srcSpotBoxs;
+    std::vector< std::shared_ptr<SpotBox> > dstSpotBoxs;
+    int fullImgWidth = 0;
+    int fullImgHeight = 0;
+    imgsrc->getFullSize(fullImgWidth, fullImgHeight, tr);
+    SpotBox fullImageBox(0, 0, fullImgWidth - 1, fullImgHeight - 1, 0, 0, nullptr, SpotBox::Type::FINAL);
+    SpotBox cropBox(pp.getX(), pp.getY(),
+                    pp.getX() + pp.getWidth() - 1, pp.getY() + pp.getHeight() - 1,
+                    0, 0, img, SpotBox::Type::FINAL);
+
+    std::set<int> visibleSpots;   // list of dest spots intersecting the preview's crop
+    int i = 0;
+
+    for (auto entry : params->spot.entries) {
+        std::shared_ptr<SpotBox> srcSpotBox(new SpotBox(entry,  SpotBox::Type::SOURCE));
+        std::shared_ptr<SpotBox> dstSpotBox(new SpotBox(entry,  SpotBox::Type::TARGET));
+        if (   !srcSpotBox->setIntersectionWith(fullImageBox)
+            || !dstSpotBox->setIntersectionWith(fullImageBox)
+            || !srcSpotBox->imageIntersects(*dstSpotBox, true))
+        {
+            continue;
+            ++i;
```

Comment:
Seems fishy - LGTM warns about this, but I have no idea about the logic of this code yet.

Reply:
It renders correctly, but if I swap the two statements, it can cause a crash. I will remove the dead code.